livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/stt.py

```diff
@@ -130,7 +177,102 @@ def with_groq(
             language=language,
             detect_language=detect_language,
             prompt=prompt,
+            use_realtime=False,
+        )
+
+    def stream(
+        self,
+        *,
+        language: str | None = None,
+        conn_options: APIConnectOptions = DEFAULT_API_CONNECT_OPTIONS,
+    ) -> "SpeechStream":
+        config = self._sanitize_options(language=language)
+        stream = SpeechStream(
+            stt=self,
+            opts=config,
+            pool=self._pool,
         )
+        self._streams.add(stream)
+        return stream
+
+    def update_options(
+        self,
+        *,
+        model: STTModels | GroqAudioModels | str | None = None,
+        language: str | None = None,
+        prompt: str | None = None,
+        vad_threshold: float | None = None,
+        prefix_padding_ms: int | None = None,
+        silence_duration_ms: int | None = None,
+        noise_reduction_type: str | None = None,
+    ) -> None:
+        self._opts.model = model or self._opts.model
+        self._opts.language = language or self._opts.language
+        self._opts.prompt = prompt or self._opts.prompt
+        self._opts.vad_threshold = vad_threshold or self._opts.vad_threshold
+        self._opts.prefix_padding_ms = prefix_padding_ms or self._opts.prefix_padding_ms
+        self._opts.silence_duration_ms = (
+            silence_duration_ms or self._opts.silence_duration_ms
+        )
+        self._opts.noise_reduction_type = (
+            noise_reduction_type or self._opts.noise_reduction_type
+        )
+
+        for stream in self._streams:
+            stream.update_options(language=language or self._opts.language)
+
+    async def _connect_ws(self) -> aiohttp.ClientWebSocketResponse:
+        realtime_config: dict[str, Any] = {
+            "type": "transcription_session.update",
+            "session": {
+                "input_audio_format": "pcm16",
+                "input_audio_transcription": {
+                    "model": self._opts.model,
+                    "prompt": self._opts.prompt or "",
+                    "language": self._opts.language or "",
+                },
+                "turn_detection": {
+                    "type": "server_vad",
```

Comment:
Do you think we can use semantic_vad? or it is the same label now?

Reply:
we are not relying on this for turn detection, so focusing on a simple, but predictable behavior here.

Reply:
but agreed we should test it and see how it performs